citrix_receiver: allow custom certs and hooks for the installation

Ma27 opened the pull request:
###### Motivation for this change

Sometimes it's required to modify some parts of the Citrix build on
their own which is why `{pre,post}Install` hooks can be quite helpful.

Additionally some corporate clients use their own certificates that
aren't stored as trusted ones in the `cacert` package with all of the
trusted certs by Mozilla.

Now it's possible to add custom certs like this:

``` nix
with import <nixpkgs> { config.allowUnfree = true; };

let path = ../../Downloads/custom-corporate-cert.pem; in
citrix_receiver.override {
  extraCerts = [ path ];
}
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ma27 committed:
citrix_receiver: allow custom certs and hooks for the installation

Sometimes it's required to modify some parts of the Citrix build on
their own which is why `{pre,post}Install` hooks can be quite helpful.

Additionally some corporate clients use their own certificates that
aren't stored as trusted ones in the `cacert` package with all of the
trusted certs by Mozilla.

Now it's possible to add custom certs like this:

``` nix
with import <nixpkgs> { config.allowUnfree = true; };

let path = ../../Downloads/custom-corporate-cert.pem; in
citrix_receiver.override {
  extraCerts = [ path ];
}
```

Ma27 commented:
Note: I'm currently testing against older Citrix versions, it currently works with 13.10 though.

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44522&attempt_id=00d69d69-9380-4526-9bdc-8c65ebcb21f0)

The following builds were skipped because they don't evaluate on aarch64-linux: citrix_receiver

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44522&attempt_id=31b38b53-d83f-49c5-818f-c5cccb1612b0)

The following builds were skipped because they don't evaluate on x86_64-linux: citrix_receiver

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

Ma27 commented:
just tested my patch with the oldest citrix version available in nixpkgs (13.4.0) and it works fine as well

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44522&attempt_id=0fa336fc-996f-456c-b580-15b07429acd1)

The following builds were skipped because they don't evaluate on aarch64-linux: citrix_receiver

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44522&attempt_id=890a9308-6893-417a-9466-c4ff3f564abf)

The following builds were skipped because they don't evaluate on x86_64-linux: citrix_receiver

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

Mic92 commented on pkgs/applications/networking/remote/citrix-receiver/default.nix:
```diff
@@ -156,7 +156,7 @@ let
       };
 
-      phases = [ "unpackPhase" "installPhase" ];
+      phases = [ "unpackPhase" "installPhase" "fixupPhase" ];
```
`dontBuild = true;` seems more economical here.

Mic92 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1735,15 +1735,43 @@ with pkgs;
+  citrix_receiver_unwrapped_13_7_0  = citrix_receiver_unwrapped.override { version = "13.7.0";  };
+  citrix_receiver_unwrapped_13_6_0  = citrix_receiver_unwrapped.override { version = "13.6.0";  };
+  citrix_receiver_unwrapped_13_5_0  = citrix_receiver_unwrapped.override { version = "13.5.0";  };
+  citrix_receiver_unwrapped_13_4_0  = citrix_receiver_unwrapped.override { version = "13.4.0";  };
```
For the next release we should drop versions older then 13.8, since there support will be dropped before 19.03: https://www.citrix.co.uk/support/product-lifecycle/milestones/receiver.html

Ma27 committed:
citrix_receiver: drop all packages below 13.8

All Citrix versions until 13.7 will be EOLed during the lifetime of
18.09: https://www.citrix.co.uk/support/product-lifecycle/milestones/receiver.html

GrahamcOfBorg commented:
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44522&attempt_id=06f8381e-eacb-4784-ad5f-3f3af782257d)

The following builds were skipped because they don't evaluate on x86_64-linux: citrix_receiver

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44522&attempt_id=404594d2-c870-4dfc-92a7-d8d0ced2685f)

The following builds were skipped because they don't evaluate on aarch64-linux: citrix_receiver

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>
